lib/tests/test_local_working_copy.rs

```diff
@@ -583,28 +583,22 @@ fn test_materialize_conflicted_files() {
         }
     );
 
```

Comment:
I don't think this is urgent, but perhaps worthy of a TODO: we could demo that conflicts are not simplified per-hunk. So, if there was only one file in this example and the different conflicts happened in different parts of it, there would be no simplification.

As I mentioned in https://github.com/martinvonz/jj/pull/3719#issuecomment-2141001204, I think this is what we want, but it's worth pointing out.